subsys/usb/class/msc.c

```diff
@@ -260,47 +260,36 @@ static bool write(uint8_t *buf, uint16_t size)
  *
  * @return  0 on success, negative errno code on fail.
  */
-static int mass_storage_class_handle_req(struct usb_setup_packet *pSetup,
+static int mass_storage_class_handle_req(struct usb_setup_packet *setup,
 					 int32_t *len, uint8_t **data)
 {
-	if (pSetup->wIndex != mass_cfg.if0.bInterfaceNumber ||
-	    pSetup->wValue != 0) {
-		LOG_WRN("Invalid setup parameters");
+	if (setup->wIndex != mass_cfg.if0.bInterfaceNumber ||
+	    setup->wValue != 0) {
+		LOG_ERR("Invalid setup parameters");
 		return -EINVAL;
 	}
 
-	switch (pSetup->bRequest) {
-	case MSC_REQUEST_RESET:
-		LOG_DBG("MSC_REQUEST_RESET");
-
-		if (pSetup->wLength) {
-			LOG_WRN("Invalid length");
-			return -EINVAL;
+	if (REQTYPE_GET_DIR(setup->bmRequestType) == REQTYPE_DIR_TO_DEVICE) {
+		if (setup->bRequest == MSC_REQUEST_RESET &&
+		    setup->wLength == 0) {
```

Comment:
Until now, if `wLength != 0` `-EINVAL` was returned. With this change `-ENOTSUP`. Does this mater? I think no, but i want to confirm that.

Reply:
It does not matter, fact is that -EINVAL was never implemented or thought through.
Comment about -EINVAL should actually be removed from usb_device.h